src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpEditing.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.VisualCSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpEditing : AbstractEditorTest
+    {
+        private const string TestSource = @"using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ConsoleApplication1
+{
+    $$public class Program
+    {
+        /// <summary/>
+        public static void Main(string[] args)
+        {
+            /* Console.WriteLine(""Uncomment"");*/
+            Console.WriteLine(""Hello World"");
+        }
+    }
+}";
+
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpEditing(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpEditing))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void IndentUnindentDoNotIntroduceErrors()
+        {
+            SetUpEditor(TestSource);
+
+            VisualStudio.ExecuteCommand("Edit.SelectAll");
+            VisualStudio.ExecuteCommand("Edit.InsertTab");
+            VisualStudio.Workspace.WaitForAllAsyncOperations();
```

Comment:
No waiting directly in tests. There shouldn't be a need to wait anyways since the ExecuteCommand is synchronous.